package-examples/wordpress-with-dependencies/deployment/deployment.yaml

```diff
@@ -0,0 +1,50 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: wordpress
+  namespace: default
+  labels:
+    app: wordpress
+  annotations:
+    config.kubernetes.io/depends-on: apps/namespaces/default/StatefulSet/wordpress-mysql
```

Comment:
Should `set-namespace` and `ensure-name-substring` be aware of these annotations ? 
To clarify the use-case, so I have a kpt pkg that have resources annotated with `depends-on` annotation. The pkg also uses functions that manipulates the namespace or name, should those function also update the `depends-on` annotation ?

/cc @mengqiy

Reply:
Yeah, that is a great point. We should see if we can add support for the depends-on annotation in the functions. I have created an issue for it: https://github.com/GoogleContainerTools/kpt/issues/2579